tfx/orchestration/pipeline.py

```diff
@@ -189,3 +200,23 @@ def components(self, components: List[base_node.BaseNode]):
     # has all its dependencies visited.
     if len(self._components) < len(deduped_components):
       raise RuntimeError('There is a cycle in the pipeline')
+
+  def set_executor(self, component_id: Text, executor_factory: Type[FakeComponentExecutorFactory]) -> None:
+    for component in self._components:
+      if component_id is component.id:
+        self.mock_executor_spec[component_id] = executor_factory()
```

Comment:
How about assigning executor_spec directly here? For example,
```
component.executor_spec = some_executor_spec
```
Then we might not need to add a new argument for mock executors in other APIs.

Reply:
+1.
This will work.
component.executor_spec = ExecutorClassSpec(executor_class)